src/7-to-8/major-changes/suicide-triggers.rst

```diff
@@ -1,64 +1,57 @@
-.. _728.suicide_triggers:
+.. TODO: rename to "Graph branching, optional outputs and suicide triggers", and give example (the one I posted on Element)
 
-Suicide Triggers
-================
+.. _728.optional_outputs:
 
-.. admonition:: Does This Change Affect Me?
-   :class: tip
-
-   Suicide triggers are written using an exclamation mark ``!`` like so:
-
-   .. code-block:: cylc-graph
-
-      task_1 => ! task_2
+Graph branching, optional outputs and suicide triggers
+======================================================
 
-   Read this section if your workflows contain suicide triggers.
+Cylc 8 has a :ref:`new scheduling algorithm <728.scheduling_algorithm>` and
+a new syntax for dealing with tasks that may not necessarily complete.
+It handles :term:`graphs <graph>` in an event-driven manner which means
+that a workflow can follow different paths in different eventualities (without
+the need for suicide triggers). This is called :term:`graph branching`.
 
-Overview
---------
+.. admonition:: Does This Change Affect Me?
+   :class: tip
 
-In Cylc 7 :term:`suicide triggers <suicide trigger>` are used to remove
-tasks from the graph automatically at runtime.
+   This change affects you if you are upgrading a Cylc 7 workflow that
+   contains graph branches that are not necessarily expected to complete
+   at runtime. You might get a ``GraphParseError`` during validation with
+   Cylc 8.
 
-Cylc 8 handles :term:`graphs <graph>` in an event-driven manner which means
-that a workflow can follow different paths in different eventualities without
-the need for suicide triggers. This is called :term:`graph branching`.
+   Typically this will be the case if you are using any of the following in
+   the workflow graph:
 
-.. seealso::
+   - :term:`suicide triggers <suicide trigger>`
+   - the ``|`` (OR) operator
+   - the ``:fail`` output or custom ouputs
```

Comment:
These two are really just consequences of the former:

* Use of the OR operator does not mean there is graph branching.
  * e.g. `a | b => c` means `c` runs when the first of `a` or `b` succeed, it does not necessarily infer branching.
* At Cylc 7 use of the `:fail` output must always be accompanied by a suicide trigger.
   * Because failed tasks would not leave the pool unless forcefully ejected.

So I think this boils down to just suicide triggers?

Trying to come up with an example which doesn't?

Reply:
I've only come up with one example so far:

```
x:succeed | x:fail => y
```

This is equivalent to:

```
x:finish => y
```

Which is still valid at Cylc 8.

Reply:
Ah yes, it's those psudo-outputs which cause the issue:

```
a:finish => b
a => c  # ERROR
```

This will also apply to the `:finish-any` and `:finish-all` family triggers.

Reply:
I'll just mention suicide triggers then.

I was playing around with `:finish` and found a possible bug in stall behaviour for back-compat mode: https://github.com/cylc/cylc-flow/issues/4965

Reply:
Ooh, good catch, that's nasty.

Reply:
From a quick grep there are quite a lot of `:finish` triggers out there, to be expected. I think this is the only non-suicide case (?) but it is a biggie!

Reply:
> At Cylc 7 use of the :fail output must always be accompanied by a suicide trigger.
> -  Because failed tasks would not leave the pool unless forcefully ejected.

I wouldn't say that's true - manual removal of failed tasks, if they don't need to be retriggered, is pretty common.

Reply:
> Ooh, good catch, that's nasty.

Nasty bug fixed, phew.